src/device/insteonDeviceSubscribe/insteonDeviceSubscribe.ts

```diff
@@ -45,102 +62,43 @@ function onReady(node: DeviceSubscribeNode, text: string){
-	node.deviceConfigNode?.device?.on(['battery', '**' ], function (this: any, p){ onBattery(node, this.event, p) });
-	node.deviceConfigNode?.device?.on(['motion', '**' ], function (this: any, p){ onMotion(node, this.event, p) });
-	node.deviceConfigNode?.device?.on(['light', '**' ], function (this: any, p){ onLight(node, this.event, p) });
-}
-
-function onPacket(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({topic: 'packet', payload: packet});
-}
+	// Adding event listener
+	node.deviceConfigNode?.device?.on(['**'], onPacket);
 
-function onSwitch(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({
-		topic: event[0],
-		payload: {
-			status: event[1],
-			type: event[2],
-		},
-		event,
-		packet,
+	// remove the event when the node is closed
+	node.on('close', function(removed: boolean, done: () => void){
+		// Remove event listeners
+		node.deviceConfigNode?.device?.removeListener(['**'], onPacket);
+		debug(`onClose removed listeners`);
+		done();
 	});
-}
 
-function onDim(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({
-		topic: event[0],
-		payload: {
-			status: event[1],
-		},
-		event,
-		packet,
-	});
-}
-
-function onSensor(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({
-		topic: event[0],
-		payload: {
-			status: event[1],
-		},
-		event,
-		packet,
-	})
-}
-
-function onHeartbeat(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({
-		topic: event[0],
-		payload: {
-			status: event[1],
-		},
-		event,
-		packet,
-	})
-}
-
-function onBattery(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({
-		topic: event[0],
-		payload: {
-			status: event[1],
-		},
-		event,
-		packet,
-	})
-}
-
-function onMotion(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({
-		topic: event[0],
-		payload: {
-			status: event[1],
-		},
-		event,
-		packet,
-	})
-}
-
-function onLight(node: DeviceSubscribeNode, event: string[], packet: Packet.Packet){
-	node.send({
-		topic: event[0],
-		payload: {
-			status: event[1],
-		},
-		event,
-		packet,
-	})
+	function onPacket(this: any, packet: Packet.Packet){
+		const event = this.event;
+		const type = this.event.pop(); // The remote/physical type is always the last part of the event
+		
+		/* Determine whether to send based on if the event type (all/physical/remote) matches what the user subscribed to */
+		if(node.subtype !== "" && node.subtype !== type) return;
+		
+		/* now filter based on the specific event types that were selected by the user
+		 * the GUI has switched.on.fast, so we join the event array by dots and compare the strings
+		 */
+		
+		if(node.selectedEvents.filter((e: any) => e.event === event.join('.')).length === 1){
+			event.push(type); // put the type back on the event array
+			node.send({
+				topic: event.join('.'),
+				payload: 'device level tbd',
```

Comment:
"device level tbd"?

Reply:
It's TBD because it's needed to make the node useful, but it requires functionality from the underlying `insteon-plm` library that isn't there yet.

I wanted to check with you what the best way to approach that would be.

I think to include the device's `level` in the packet we first have to change the packet type and the InsteonDevice type to eacy include a `level` property.

To populate the level on the device object we can include a routine whenever a command is sent to the device that calculates the level based on the command itself - e.g. `[0x11, 0xFF]` would just set the level to `0xFF`, while BrightenOneStep would have to add 8 to the current level.

When a device responds to a scene on command, we could take the device's responder onLevel.